packages/runtime/plugin-i18n/src/runtime/I18nLink.tsx

```diff
@@ -0,0 +1,83 @@
+import type React from 'react';
+import { useModernI18n } from './context';
+import { buildLocalizedUrl } from './utils';
+
+export interface I18nLinkProps {
+  to: string;
+  children: React.ReactNode;
+  [key: string]: any; // Allow other props to be passed through
+}
+
+/**
+ * I18nLink component that automatically adds language prefix to navigation links.
+ * This component should be used within a :lang dynamic route context.
+ *
+ * @example
+ * ```tsx
+ * // When current language is 'en' and to="/about"
+ * // The actual link will be "/en/about"
+ * <I18nLink to="/about">About</I18nLink>
+ *
+ * // When current language is 'zh' and to="/"
+ * // The actual link will be "/zh"
+ * <I18nLink to="/">Home</I18nLink>
+ * ```
+ */
+// Safe hook wrapper for router hooks
+const useRouterHooks = () => {
+  try {
+    const { Link, useParams } = require('@modern-js/runtime/router');
```

Comment:
It will cause the router's tree-shaking to fail, do we can use `useInRouterContext` here?